editoast/src/infra.rs

```diff
@@ -208,6 +208,29 @@ impl Infra {
         }
     }
 
+    pub fn clone(infra_id: i64, conn: &mut PgConnection) -> Result<Infra, Box<dyn ApiError>> {
+        let infra_to_clone = Infra::retrieve(conn, infra_id)?;
+        let infra_name = infra_to_clone.name + "_Copy";
```

Comment:
What about checking that the infra name doesn't already finish by `(Copy)`.
Another solution is to add a query param `infra_name` that is required and the front should handle how to name the duplicated infra.

```suggestion
        let infra_name = infra_to_clone.name + " (Copy)";
```